src/librustc_metadata/schema.rs

```diff
@@ -200,7 +200,7 @@ pub struct TraitImpls {
 #[derive(RustcEncodable, RustcDecodable)]
 pub struct Entry<'tcx> {
     pub kind: EntryKind<'tcx>,
-    pub visibility: ty::Visibility,
+    pub visibility: Lazy<ty::Visibility>,
```

Comment:
I had to change this to `Lazy` so that we don't try to decode a `DefId` before we have access to the `cnum_map` (causing a panic).

cc @eddyb